### PR TITLE
Publish docs only for releases and update GHA

### LINKED ISSUE #116: Docs not sync'ed with current release
Docs state version 1.2.7: 
image

Looks like most recent release is 1.2.6: 
image

Looks like 1.2.7 is a decent change from 1.2.6:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In [9], line 2
      1 from pyobis import dataset
----> 2 query = dataset.get(dataset_ids[0])
      3 query.execute() # execute the query
      4 # query.data # returns the data of the query
      5 # query.api_url # returns the OBIS API URL

AttributeError: module 'pyobis.dataset.dataset' has no attribute 'get'
```

Consider a larger version increment for these changes? 1.3.0??

### PR DESCRIPTION
**Pull request by ocefpaf**
Alternative for #117 and fixes #116.

### CODE REVIEW

File: .github/workflows/deploy-docs.yml [hunk 1]
```diff
@@ -1,10 +1,13 @@
 name: Build and Deploy docs
 
 on:
+  pull_request:
```
Comment (ocefpaf) [inline .github/workflows/deploy-docs.yml:4]: @ayushanand18 we want to run in all PRs and merges to main to ensure the docs are building but we do not want to publish but we only want to publish when releasing.

Comment (ayushanand18) [reply .github/workflows/deploy-docs.yml:4]: Gotcha! I understand the workflow now 👀

**Comment by ayushanand18** [comment]
Thank you so much @ocefpaf! But we would like to fast-forward to `1.3.0`, I understand that whenever we publish the docs, the version gets incremented from the current version but how do we fast forward it to `1.3.0` or define if we have a `major`/`minor`/`patch` update and increment the version accordingly?

File: .github/workflows/deploy-docs.yml (hunk 1 above)
Comment (ocefpaf) [reply .github/workflows/deploy-docs.yml:4]: It took me a while to understand all that. I created this problem in the original PR BTW :flushed:

**Review by ayushanand18 — APPROVED** [review]

**Comment by ocefpaf** [comment]
> Thank you so much @ocefpaf! But we would like to fast-forward to `1.3.0`, I understand that whenever we publish the docs, the version gets incremented from the current version but how do we fast forward it to `1.3.0` or define if we have a `major`/`minor`/`patch` update and increment the version accordingly?

All we need to do is to tag it 1.3.0. When this is ready, tag it and it will propagate everywhere. The "dirty" git dev tag doesn't increment much b/c it only know that it is different from the last tagged versions. Also, we don't really care about the dirty dev one that much, it is only used do ensure we know the git commit used when debugging it.

**Comment by ayushanand18** [comment]
> All we need to do is to tag it 1.3.0. When this is ready, tag it and it will propagate everywhere. The "dirty" git dev tag doesn't increment much b/c it only know that it is different from the last tagged versions. Also, we don't really care about the dirty dev one that much, it is only used do ensure we know the git commit used when debugging it.

I got it, thanks! This is some amazing engineering that will take some iterations for me to completely understand 👀

**Comment by MathewBiddle** [comment]
Something of interest. A `conda search pyobis` returns the two different versions, but it looks like the builds are the same? That's probably okay and something I need to learn about?

```
>conda search pyobis
Loading channels: done
# Name                       Version           Build  Channel             
pyobis                         1.2.6    pyhd8ed1ab_0  conda-forge         
pyobis                         1.3.0    pyhd8ed1ab_0  conda-forge    
```

**Comment by ocefpaf** [comment]
> but it looks like the builds are the same?

That is either a bug in conda-build or a super coincidence! The build hash should be unique to each build, even of the same version when we change things. With that said, it is not a big deal. I checked the code and the versions are different and the right ones.

**Comment by 7yl4r** [comment]
> super coincidence

a super duper extra rare crazy unlikely miracle coincidence

**Pull request merged**